### DISCUSSION TITLE
Sequential Agent is unable to use value from Set/Get Variable

### DISCUSSION THREAD

**Discussion by tarkancorak** [FlowiseAI/Flowise#2947] [Q&A]
Does anybody know if it is by intention that the Sequential Agent does not offer a selection for a previously set variable? I'm setting a value with a Custom JS Function & Set Variable. When I connect it to the Format Prompt Values of the Sequential Agent and try to access the value from Set Variable, it does not provide me this option. However, when I try it out with an LLM Chain, it works. But for Agents, AFAIK, there is no option to use an LLM Chain.

I would like to know if there is another option to use variables in such an agentic scenario. And it would already help to know if this is actually a bug, or to know the reason why this option is not available. Thanks in advance!

Format Prompt Values for a Prompt Template:
![Format Prompt Values for a Prompt Template](https://github.com/user-attachments/assets/cfa7f6ed-d159-412c-9b96-6f5a5da426a8)

Format Prompt Values for a Sequential Agent
![Format Prompt Values for a Sequential Agent](https://github.com/user-attachments/assets/a81002bf-6e7a-4216-809b-ad5ce8cba465)

**Comment by bernejorge** [comment]
I think that you can use states as another option
